Source/WebKit/UIProcess/BrowsingContextGroup.cpp

```diff
@@ -88,13 +88,14 @@ void BrowsingContextGroup::addFrameProcess(FrameProcess& process)
     auto& site = process.site();
     ASSERT(site.isEmpty() || !m_processMap.get(site) || m_processMap.get(site)->process().state() == WebProcessProxy::State::Terminated || m_processMap.get(site) == &process);
     m_processMap.set(site, process);
-    for (auto& page : m_pages) {
-        if (site == Site(URL(page.currentURL())))
+    for (Ref page : m_pages) {
+        if (site == Site(URL(page->currentURL())))
             return;
         auto& set = m_remotePages.ensure(page, [] {
             return HashSet<Ref<RemotePageProxy>> { };
         }).iterator->value;
-        Ref newRemotePage = RemotePageProxy::create(page, process.process(), site);
+        Ref processProxy = process.process();
```

Comment:
Ideally we'd move this before the loop to avoid refing/derefing the same process for each loop iteration.

Reply:
Good point. Done.